crypto-ffi/src/error/core_crypto.rs

```diff
@@ -67,25 +67,55 @@ impl From<RecursiveError> for CoreCryptoError {
         /// Hopefully only ever use this in conjunction with `interior_matches!`, because for most sane
         /// circumstances, `if let` is the better design pattern.
         macro_rules! matches_option {
-            ($val:expr, $pattern:pat $(if $guard:expr)? => $out:expr) => {
+            // Without cfg attribute
+            (
+                $val:expr,
+                $pattern:pat $(if $guard:expr)? => $out:expr
+            ) => {
                 match ($val) {
                     $pattern $(if $guard)? => Some($out),
                     _ => None,
                 }
             };
+            // With cfg attribute
+            (
+                $val:expr,
+                #[cfg($meta:meta)]
```

Comment:
I approve of the restraint here, not expanding an arbitrary `$( #[$meta:meta] )*`; YAGNI.